### PR TITLE
support customized network in node role

### PR DESCRIPTION
**Pull request by jhd2best**
support customized network by use Ansible -extra-var param **network**, and default is mainnet

### CODE REVIEW

**Comment by jhd2best** [comment]
**Test Result**
PASS

**Test Log**
ansible-playbook -i inventory/haodi.hosts playbooks/install-node.yml -e 'inventory=test user=hmy network=staking' --tags="test,vars"

PLAY [test] *******************************************************************************************************************************************************

TASK [Gathering Facts] ***********************************************************************************************************************************************
ok: [52.\*.\*.139]

TASK [node : gather OS specific variables] ***************************************************************************************************************************
ok: [52.\*.\*.139] => (item=/experiment-deploy/ansible/playbooks/roles/node/vars/Azure-CentOS.yml)

TASK [node : copy harmony service files] *****************************************************************************************************************************
changed: [52.\*.\*.139]

PLAY RECAP ***********************************************************************************************************************************************************
52.\*.\*.139             : ok=3    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

**Content Of harmony.service**
[Unit]
Description=harmony service
After=network.target
StartLimitIntervalSec=0

[Service]
Type=simple
Restart=always
RestartSec=1
User=hmy
WorkingDirectory=/home/hmy
ExecStart=/home/hmy/node.sh -N staking -p /home/hmy/.hmy/blskeys/bls.pass -T explorer -i 0 -A -M -D -P -1

[Install]
WantedBy=multi-user.target

File: ansible/playbooks/roles/node/templates/harmony.service.j2 [hunk 1]
```diff
@@ -0,0 +1,23 @@
+[Unit]
+Description=harmony service
+After=network.target
+StartLimitIntervalSec=0
```
Comment (LeoHChen) [inline ansible/playbooks/roles/node/templates/harmony.service.j2:4]: this is for systemd > 219 version, on AWS the systemd version is older, and can't support this variable, it is called `StartLimitInterval`

File: ansible/playbooks/roles/node/templates/harmony.service.j2 [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+{% if 'exp' in inventory %}
+ExecStart={{ USER.home }}/node.sh -N {{ network }} -p {{ USER.home }}/.hmy/blskeys/bls.pass -T explorer -i 0 -A -S -M -D -P -1
+{% else %}
+{% if network in ['staking'] %}
+ExecStart={{ USER.home }}/node.sh -z -N {{ network }} -p {{ USER.home }}/.hmy/blskeys/bls.pass -S -M -D -P -1
+{% else %}
+ExecStart={{ USER.home }}/node.sh -N {{ network }} -p {{ USER.home }}/.hmy/blskeys/bls.pass -S -M -D -P -1
+{% endif %}
+{% endif %}
+
```
Comment (LeoHChen) [inline ansible/playbooks/roles/node/templates/harmony.service.j2:21]: add standarderr etc to this template, check this one, https://github.com/harmony-one/experiment-deploy/blob/master/tools/harmony.service.template

File: ansible/playbooks/roles/node/tasks/main.yml [hunk 3]
```diff
@@ -6,6 +6,7 @@
     - '{{ ansible_distribution }}-{{ ansible_distribution_major_version }}.yml'
     - '{{ ansible_distribution }}.yml'
     - '{{ ansible_os_family }}.yml'
+    - '{{ cloud_platform }}-{{ ansible_distribution }}.yml'
```
Comment (LeoHChen) [inline ansible/playbooks/roles/node/tasks/main.yml:9]: good one, can we change our aws/gcp yml files using this format?

File: ansible/playbooks/roles/node/templates/harmony.service.j2 (hunk 1 above)
Comment (jhd2best) [reply ansible/playbooks/roles/node/templates/harmony.service.j2:4]: got it. i will make it compatible

File: ansible/playbooks/roles/node/templates/harmony.service.j2 (hunk 2 above)
Comment (jhd2best) [reply ansible/playbooks/roles/node/templates/harmony.service.j2:21]: already added

File: ansible/playbooks/roles/node/tasks/main.yml (hunk 3 above)
Comment (jhd2best) [reply ansible/playbooks/roles/node/tasks/main.yml:9]: sure. but need collect aws/gcp's ansible_distribution content in advance

**Pull request merged**